src/main/java/zd/zero/waifu/motivator/plugin/listeners/WaifuUnitTesterImpl.java

```diff
@@ -1,54 +1,68 @@
-import com.intellij.notification.Notification;
-import com.intellij.notification.NotificationType;
-import com.intellij.notification.Notifications;
+import com.intellij.execution.testframework.sm.runner.SMTRunnerEventsAdapter;
+import com.intellij.execution.testframework.sm.runner.SMTRunnerEventsListener;
+import com.intellij.execution.testframework.sm.runner.SMTestProxy;
+import com.intellij.openapi.externalSystem.service.execution.ExternalSystemProcessHandler;
 import com.intellij.util.messages.MessageBusConnection;
 import org.jetbrains.annotations.NotNull;
-import zd.zero.waifu.motivator.plugin.tools.Debouncer;
+import zd.zero.waifu.motivator.plugin.listeners.adapter.WaifuTestPrinterAdapter;
 
 public class WaifuUnitTesterImpl implements WaifuUnitTester {
 
     private final MessageBusConnection busConnection;
 
     private final WaifuUnitTester.Listener listener;
 
-    private final Debouncer debouncer;
-
     public WaifuUnitTesterImpl( MessageBusConnection busConnection,
-                                Listener listener,
-                                Debouncer debouncer ) {
+                                Listener listener ) {
         this.busConnection = busConnection;
         this.listener = listener;
-        this.debouncer = debouncer;
     }
 
     @Override
     public void init() {
-        busConnection.subscribe( Notifications.TOPIC, new Notifications() {
+        busConnection.subscribe( SMTRunnerEventsListener.TEST_STATUS, new SMTRunnerEventsAdapter() {
             @Override
-            public void notify( @NotNull Notification notification ) {
-                invokeListener( notification, TestsUIUtil.NOTIFICATION_GROUP.getDisplayId() );
-            }
-        } );
-    }
-
-    void invokeListener( Notification notification, final String NOTIFICATION_GROUP_DISPLAY_ID ) {
-        if ( notification.getGroupId().equals( NOTIFICATION_GROUP_DISPLAY_ID ) ) {
-            debouncer.debounce( () -> {
-                if ( notification.getType() == NotificationType.ERROR ) {
-                    listener.onUnitTestFailed();
-                } else {
+            public void onTestingFinished( @NotNull SMTestProxy.SMRootTestProxy testsRoot ) {
+                if ( testsRoot.wasTerminated() || testsRoot.isInterrupted()
+                    || isCancelledOnExternalProcess( testsRoot ) ) return;
+                if ( testsRoot.isPassed() ) {
                     listener.onUnitTestPassed();
+                } else {
+                    listener.onUnitTestFailed();
                 }
-            } );
-        }
+            }
+        } );
     }
 
     @Override
     public void stop() {
         this.busConnection.disconnect();
     }
 
+    /**
+     * <p>A compatibility hack method to perform checking
+     * for external processes (Gradle test execution)
+     * to check if user has cancelled the test.</p>
+     *
+     * <p>For <code>202.4357.23</code>, it should be replaced with</p>
```

Comment:
When do we plan to only support this version and above? Right now almost all of the 2020.2 products are above this version (I think). I know Android Studio is still behind.

Reply:
I noticed that other users don't update their IDEs as other plugins are not supported, with that in mind, for the minimum support version, I'm thinking if we start bumping it for `v1.3`, or after a couple of weeks.
Do you suggest to do it now?

Reply:
> Do you suggest to do it now?

Nope, I think if you want the plugin available to a wider audience then maintaining a wider build support helps. However, I do think that eventually making `202.4357.23` the minimum version would be good to do in a near future release.